File: typescript/Creational/01-FactoryMethod/Creator/JobCardCreator.ts
abstract class JobCardCreator {
  // The Factory Method
  abstract createJobCard(title: string, priority: string): JobCard;

  // Template method — uses the factory method internally
  issueJobCard(title: string, priority: string): JobCard {
    const card = this.createJobCard(title, priority);
    console.log("✅ Job card issued by FixIt Pro dispatch.");
    card.printSummary();
    return card;
  }
}

File: csharp/Creational/01-FactoryMethod/Creator/JobCardCreator.cs
public abstract class JobCardCreator
{
    // The Factory Method
    public abstract IJobCard CreateJobCard(string title, string priority);

    // Template method — uses the factory method internally
    public IJobCard IssueJobCard(string title, string priority)
    {
        var card = CreateJobCard(title, priority);
        Console.WriteLine("✅ Job card issued by FixIt Pro dispatch.");
        card.PrintSummary();
        return card;
    }
}